### PR TITLE
feat: [STO-5050]: Bandit Security Step

### PR DESCRIPTION
**Pull request by jsentongo**
#### Screenshots

https://user-images.githubusercontent.com/94563383/209703085-757ef8c7-4cba-486f-96d0-6bc4f4611f58.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by atmollo-harness — DISMISSED** [review]

**Comment by jsentongo** [comment]
retrigger prettier

File: src/modules/85-sto/components/PipelineSteps/BanditStep/BanditStep.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+  onUpdate?: (data: BanditStepData) => void
+  onChange?: (data: BanditStepData) => void
+  allowableTypes: AllowedTypes
+  formik?: any
+}
+
+export class BanditStep extends PipelineStep<BanditStepData> {
+  constructor() {
+    super()
+    this._hasStepVariables = true
```
Comment (ishantmehtaharness) [inline src/modules/85-sto/components/PipelineSteps/BanditStep/BanditStep.tsx:43]: Do you want `this._hasDelegateSelectionVisible = true` also?

File: src/modules/85-sto/components/PipelineSteps/SecurityFields.tsx [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+      stepViewType={stepViewType}
+      allowableTypes={allowableTypes}
+      formik={formik as unknown as FormikProps<T>}
+      enableFields={{
+        'spec.advanced.log.level': {
+          fieldType: 'dropdown',
+          label: 'sto.stepField.advanced.logLevel',
+          selectItems: [
+            {
+              label: 'Info',
```
Comment (ishantmehtaharness) [inline src/modules/85-sto/components/PipelineSteps/SecurityFields.tsx:140]: We should use getString for internationalization

**Review by ishantmehtaharness — APPROVED** [review]
Looks good, with 2 comments

**Review by atmollo-harness — DISMISSED** [review]

**Comment by jsentongo** [comment]
retrigger typecheck

File: src/modules/85-sto/components/PipelineSteps/BanditStep/BanditStep.tsx (hunk 1 above)
Comment (jsentongo) [reply src/modules/85-sto/components/PipelineSteps/BanditStep/BanditStep.tsx:43]: I'll Add this in the next PR

File: src/modules/85-sto/components/PipelineSteps/SecurityFields.tsx (hunk 2 above)
Comment (jsentongo) [reply src/modules/85-sto/components/PipelineSteps/SecurityFields.tsx:140]: This too I'll take care of in the next PR.

Comment (pankajsingh-harness) [reply src/modules/85-sto/components/PipelineSteps/SecurityFields.tsx:140]: These options can be moved to a variable and then used here. Also how we are making the value compatible with the backend here. As value is string here, so any string can be accepted.

There should be one to one mapping with the value here and values used in backend.

File: src/modules/85-sto/components/PipelineSteps/SecurityFields.tsx [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+          ]
+        },
+        'spec.advanced.args.cli': {
+          label: 'sto.stepField.advanced.cli'
+        },
+        'spec.advanced.fail_on_severity': {
+          fieldType: 'dropdown',
+          label: 'sto.stepField.advanced.failOnSeverity',
+          selectItems: [
+            {
```
Comment (pankajsingh-harness) [inline src/modules/85-sto/components/PipelineSteps/SecurityFields.tsx:164]: Same as above

File: src/modules/85-sto/components/PipelineSteps/types.ts [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+import type { Types } from '@pipeline/components/PipelineSteps/Steps/StepsTransformValuesUtils'
+import type { StringsMap } from 'stringTypes'
+import type { Types as ValidationFieldTypes } from '@pipeline/components/PipelineSteps/Steps/StepsValidateUtils'
+import type {
+  MultiTypeMapType,
+  MultiTypeSelectOption,
+  Resources
+} from '@pipeline/components/PipelineSteps/Steps/StepsTypes'
+
+export type ScanTool =
```
Comment (pankajsingh-harness) [inline src/modules/85-sto/components/PipelineSteps/types.ts:17]: this is not used anywhere so we can remove this as of now,

**Review by atmollo-harness — APPROVED** [review]

**Pull request merged**